docs/incentivizedtestnet/03-challenges/02-details/01-build_a_verifier.md

```diff
@@ -0,0 +1,95 @@
+* SP1
+* Jolt
+* Kimchi + Pickle
+    * Objective: Support verification of Mina’s succinct state proofs as well as proofs generated via O1-JS
+* Nova
+    * Over “pasta” cycle of elliptic curves
+* Supernova 
+    * Over “pasta” cycle of elliptic curves
+
+### Requirements and Best Practices
+
+* The programming language should be Rust, the <u><i>latest stable</i></u> version of the toolchain must be used (as to enable direct runtime inclusion via WASM compilation).
+
+* Try to leverage as much as possible already existing and possibly well audited/battle-tested solutions, in case you plan to adopt third party libraries, and make sure that such libraries have some kind of open-source license.
+
+* Add tests for the Verification Library:
+    * Tests should cover happy/unhappy paths for proof verification and serialization/deserialization of vk/proof/public inputs. 
+    * Include some tests with hardcoded data ideally taken from third-party on-chain/official sources, depending on the use case for which we wish the verifier to be integrated.
+
+
+* Add tests for the Verification Pallet. Tests should include:
+    * Correct inclusion of the pallet in the runtime.
+    * Unit tests with mock runtime.
+    * Weight tests.
+    * Modifications to e2e tests which already tests the other already included verifiers
+
+
+* Documentation for the newly added verification pallet must be added to zkverify-docs [repository](https://github.com/HorizenLabs/zkverify-docs). Please, follow the same pattern as the [ones](https://docs.zkverify.io/overview/verification_pallets/abstract/) already present.
+
+
+* An end-to-end tutorial on how to submit proofs for the verifier you’ve just added to the zkVerify blockchain must be provided. For instance, if you are integrating the gnark verifier, make sure to document or reference how to generate gnark proofs using the gnark toolchain.
+
+
+* Make sure to provide any tool that users might require to transform proofs, vk and public inputs from the chosen source (e.g. Gnark) to the format accepted by the zkVerify blockchain, as provided by your implementation.
+
+
+* For the submission itself to the zkVerify blockchain, feel free to leverage either the Polkadot JS frontend for submission via copy-paste, or any kind of Javascript/Rust code to do it programmatically. 
+
+
+* As a reference, take a look at the tutorials we already [have](https://docs.zkverify.io/tutorials/submit-proofs/typescript-example) in our documentation.
+
+### Acceptance Criteria and Submission
+
+
+* Implementations with “no-std” (thus compilable in WASM and directly includible in the runtime) are preferred over “native” ones.
+
+
+* Code must compile and CI must pass. Take a look at the [instructions](https://testnet-explorer.zkverify.io/) on how to run the CI locally for more information. If you require the CI to install some dependencies, feel free to modify it. Otherwise reach the team for further support.
```

Comment:
```suggestion
* Code must compile and CI must pass. Take a look at the [instructions](https://github.com/HorizenLabs/zkVerify?tab=readme-ov-file#running-github-workflows-on-local-environment) on how to run the CI locally for more information. If you require the CI to install some dependencies, feel free to modify it. Otherwise reach the team for further support.
```